apps/web/src/utils/domUtils.ts

```diff
@@ -2,22 +2,21 @@
  * This will paste the given text into the textarea element
  */
 export function insertText(newText: string, textarea: HTMLTextAreaElement) {
-  // make sure it's selected when we go to pasta
-  textarea.focus();
+  // Get the current cursor position
+  const cursorPos = textarea.selectionStart;
 
-  let pasted = true;
-  try {
-    if (!document.execCommand('insertText', false, newText)) {
```

Comment:
this is deprecated so moving to a modern impl

Reply:
@typesafegh is gonna figure out how to make the new impl work for a controlled component ong

Reply:
sori no ingles